dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/SiteViewPaginator.java

```diff
@@ -0,0 +1,169 @@
+package com.dotcms.rest.api.v1.secret;
+
+import com.dotcms.rest.api.v1.secret.view.SiteView;
+import com.dotcms.util.pagination.OrderDirection;
+import com.dotcms.util.pagination.PaginationException;
+import com.dotcms.util.pagination.PaginatorOrdered;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.common.model.ContentletSearch;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PaginatedArrayList;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * PaginatorOrdered implementation for objects of type SiteView.
+ * This Pagination isn't very typical in the sense that items are not retrieved from a database.
+ * Sorting and filtering itself happens right here on top of the list of elements itself.
+ */
+public class SiteViewPaginator implements PaginatorOrdered<SiteView> {
+
+    private static final String CONTENT_TYPE_HOST_QUERY = "+contentType:Host +working:true ";
+    private static final String CONTENT_TYPE_HOST_WITH_TITLE_QUERY = "+contentType:Host +working:true +title:*%s*";
+
+    private Supplier<Set<String>> configuredSitesSupplier;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Private field 'configuredSitesSupplier' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960082)